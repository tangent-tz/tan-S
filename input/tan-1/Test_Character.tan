main {
    # Test 1: Declare a char variable with a simple printable ASCII character
    var cone := 'a';
    print cone;
    print \n;

    # Expected output:
    # a

    #==================================================================

    # Test 2: Declare a char variable with octal numeric code
    var ctwo := %124;
    print ctwo;
    print \n;

    # Expected output:
    # T

    #==================================================================

    # Test 3: Re-assignment of char variable (printable ASCII -> printable ASCII)
    var cthree := 'Z';
    print cthree;
    print \n;

    cthree := '^';
    print cthree;
    print \n;

    # Expected output:
    # Z
    # ^

    #==================================================================

    # Test 4: Re-assignment of char variable (octal code -> octal code)
    var cfour := %133;
    print cfour;
    print \n;

    cfour := %176;
    print cfour;
    print \n;

    # Expected output:
    # [
    # ~

    #==================================================================

    # Test 5: Declare a const char variable
    const cfive := '=';
    print cfive;
    print \n;

    const csix := 'H';
    print csix;
    print \n;

    # Expected output:
    # =
    # H

    #==================================================================

    # Test 6: Re-assignment of char variable (printable ASCII -> octal code)
    var cseven := 'w';
    print cseven;
    print \n;

    cseven := %047;
    print cseven;
    print \n;

    # Expected output:
    # w
    # '

    #==================================================================

    # Test 7: Re-assignment of char variable (octal code -> printable ASCII)
    var ceight := %062;
    print ceight;
    print \n;

    ceight := '<';
    print ceight;
    print \n;

    # Expected output:
    # 2
    # <
}