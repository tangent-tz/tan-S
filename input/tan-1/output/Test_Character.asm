        Jump         $$main                    
        DLabel       $eat-location-zero        
        DataZ        8                         
        DLabel       $print-format-integer     
        DataC        37                        %% "%d"
        DataC        100                       
        DataC        0                         
        DLabel       $print-format-boolean     
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-float       
        DataC        37                        %% "%f"
        DataC        102                       
        DataC        0                         
        DLabel       $print-format-character   
        DataC        37                        %% "%c"
        DataC        99                        
        DataC        0                         
        DLabel       $print-format-string      
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-newline     
        DataC        10                        %% "\n"
        DataC        0                         
        DLabel       $print-format-space       
        DataC        32                        %% " "
        DataC        0                         
        DLabel       $print-format-tab         
        DataC        9                         %% "\t"
        DataC        0                         
        DLabel       $boolean-true-string      
        DataC        116                       %% "true"
        DataC        114                       
        DataC        117                       
        DataC        101                       
        DataC        0                         
        DLabel       $boolean-false-string     
        DataC        102                       %% "false"
        DataC        97                        
        DataC        108                       
        DataC        115                       
        DataC        101                       
        DataC        0                         
        DLabel       $errors-general-message   
        DataC        82                        %% "Runtime error: %s\n"
        DataC        117                       
        DataC        110                       
        DataC        116                       
        DataC        105                       
        DataC        109                       
        DataC        101                       
        DataC        32                        
        DataC        101                       
        DataC        114                       
        DataC        114                       
        DataC        111                       
        DataC        114                       
        DataC        58                        
        DataC        32                        
        DataC        37                        
        DataC        115                       
        DataC        10                        
        DataC        0                         
        Label        $$general-runtime-error   
        PushD        $errors-general-message   
        Printf                                 
        Halt                                   
        DLabel       $errors-int-divide-by-zero 
        DataC        105                       %% "integer divide by zero"
        DataC        110                       
        DataC        116                       
        DataC        101                       
        DataC        103                       
        DataC        101                       
        DataC        114                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$i-divide-by-zero        
        PushD        $errors-int-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $errors-float-divide-by-zero 
        DataC        102                       %% "float divide by zero"
        DataC        108                       
        DataC        111                       
        DataC        97                        
        DataC        116                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$f-divide-by-zero        
        PushD        $errors-float-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $usable-memory-start      
        DLabel       $global-memory-block      
        DataZ        10                        
        Label        $$main                    
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% cone
        PushI        97                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% cone
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        1                         
        Add                                    %% ctwo
        PushI        84                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        1                         
        Add                                    %% ctwo
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        2                         
        Add                                    %% cthree
        PushI        90                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        2                         
        Add                                    %% cthree
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        2                         
        Add                                    %% cthree
        PushI        94                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        2                         
        Add                                    %% cthree
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        3                         
        Add                                    %% cfour
        PushI        91                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        3                         
        Add                                    %% cfour
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        3                         
        Add                                    %% cfour
        PushI        126                       
        StoreC                                 
        PushD        $global-memory-block      
        PushI        3                         
        Add                                    %% cfour
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        4                         
        Add                                    %% cfive
        PushI        61                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        4                         
        Add                                    %% cfive
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        5                         
        Add                                    %% csix
        PushI        72                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        5                         
        Add                                    %% csix
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        6                         
        Add                                    %% cseven
        PushI        119                       
        StoreC                                 
        PushD        $global-memory-block      
        PushI        6                         
        Add                                    %% cseven
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        6                         
        Add                                    %% cseven
        PushI        39                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        6                         
        Add                                    %% cseven
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        7                         
        Add                                    %% ceight
        PushI        50                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        7                         
        Add                                    %% ceight
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        7                         
        Add                                    %% ceight
        PushI        60                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        7                         
        Add                                    %% ceight
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushI        110                       
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushI        35                        
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushI        33                        
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushI        42                        
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushI        111                       
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% cnine
        PushI        112                       
        StoreC                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% cnine
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        9                         
        Add                                    %% cten
        PushI        56                        
        StoreC                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% cnine
        PushD        $global-memory-block      
        PushI        9                         
        Add                                    %% cten
        LoadC                                  
        StoreC                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% cnine
        LoadC                                  
        PushD        $print-format-character   
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        Halt                                   
