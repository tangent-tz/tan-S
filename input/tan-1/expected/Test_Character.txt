Test 1: Declare a char variable with a simple printable ASCII character
a

Test 2: Declare a char variable with octal numeric code
T

Test 3: Re-assignment of char variable (printable ASCII -> printable ASCII)
Z
^

Test 4: Re-assignment of char variable (octal code -> octal code)
[
~

Test 5: Declare a const char variable
=
H

Test 6: Re-assignment of char variable (printable ASCII -> octal code)
w
'

Test 7: Re-assignment of char variable (octal code -> printable ASCII)
2
<

Test 8: Printing character literals directly
n
#
!
*
o

Test 9: Re-assignment of char variable from another char variable
p
8
